driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -333,11 +334,28 @@ public Session connect(String keyspace) {
      * {@link #closeAsync()}), or as a result of an error while initializing the
      * Cluster.
      */
-    public ListenableFuture<Session> connectAsync(String keyspace) {
+    public ListenableFuture<Session> connectAsync(final String keyspace) {
```

Comment:
Would be nice to update the connect / connectAsync javadoc to include a `@throws` doc for InvalidQueryException or something if setting the keyspace fails.